scala/lambdas/preingest-tdr-aggregator/src/main/scala/uk/gov/nationalarchives/preingesttdraggregator/Lambda.scala

```diff
@@ -28,13 +29,12 @@ class Lambda extends RequestHandler[SQSEvent, Unit]:
   given DASFNClient[IO] = DASFNClient[IO]()
   given DADynamoDBClient[IO] = DADynamoDBClient[IO]()
 
-  override def handleRequest(input: SQSEvent, context: Context): Unit =
-    ConfigSource.default
-      .loadF[IO, Config]()
-      .flatMap { config =>
-        Aggregator[IO].aggregate(config, groupCacheAtomicCell, input.getRecords.asScala.toList, context.getRemainingTimeInMillis)
-      }
-      .unsafeRunSync()
+  override def handleRequest(input: SQSEvent, context: Context): SQSBatchResponse = {
+    for {
+      config <- ConfigSource.default.loadF[IO, Config]()
+      results <- Aggregator[IO].aggregate(config, groupCacheAtomicCell, input.getRecords.asScala.toList, context.getRemainingTimeInMillis)
+    } yield SQSBatchResponse.builder().withBatchItemFailures(results.asJava).build
```

Comment:
What happens if there are no failures, does just send a failure message with nothing in it?

Reply:
Yep, it sends 
```json
{ 
  "batchItemFailures": []
}
```

Reply:
Should we prevent it from sending anything if there are no failures or does it not matter?

Reply:
You have to send the batch response in that format if we've configured the lambda to expect it.